### PR TITLE
Docs: Add classes to C API return value annotations

### PR DESCRIPTION
**Pull request by hugovk**
Currently light green is used for all return types 

* "Return value: Always NULL."
* "Return value: New reference."
* "Return value: Borrowed reference."

For example: https://docs.python.org/3/c-api/exceptions.html

<details>
<summary>Screenshots</summary>

image

image

image

</details>

A borrowed reference is not necessarily a "bad" thing, but we may want to use something other than green that indicates it's "good", to indicate they should be treated with care.

One option is yellow.

1. These annotations are added in `Doc/tools/extensions/c_annotations.py`. The first step is to add CSS classes in this file. 

2. The light green colour is defined in https://github.com/python/python-docs-theme. The second step is to adjust the colours in the theme.

This PR also includes some light cleanup whilst we're touching this file, see commits.

----
📚 Documentation preview 📚: https://cpython-previews--117926.org.readthedocs.build/

### CODE REVIEW

File: Doc/tools/extensions/c_annotations.py [hunk 1]
```diff
@@ -63,15 +61,15 @@ def __init__(self, name):
 class Annotations:
     def __init__(self, refcount_filename, stable_abi_file):
         self.refcount_data = {}
-        with open(refcount_filename, 'r') as fp:
+        with open(refcount_filename) as fp:
```
Comment (sobolevn) [inline Doc/tools/extensions/c_annotations.py]:
```suggestion
        with open(refcount_filename, encoding='utf8') as fp:
```

File: Doc/tools/extensions/c_annotations.py [hunk 2]
```diff
@@ -91,9 +89,8 @@ def __init__(self, refcount_filename, stable_abi_file):
                     entry.result_refs = refcount
 
         self.stable_abi_data = {}
-        with open(stable_abi_file, 'r') as fp:
+        with open(stable_abi_file) as fp:
```
Comment (sobolevn) [inline Doc/tools/extensions/c_annotations.py]:
```suggestion
        with open(stable_abi_file, encoding='utf8') as fp:
```

**Review by sobolevn — COMMENTED** [review]
👍

**Review by erlend-aasland — APPROVED** [review]
Looks good

**Review by sobolevn — APPROVED** [review]

**Comment by hugovk** [comment]
Step two: https://github.com/python/python-docs-theme/pull/188

**Pull request merged**